src/mscorlib/src/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -2,12 +2,56 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Diagnostics;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
+using Internal.Runtime.CompilerServices;
 
 namespace System.Runtime.Intrinsics
 {
     [Intrinsic]
+    [DebuggerDisplay("{DisplayString,nq}")]
+    [DebuggerTypeProxy(typeof(Vector128DebugView<>))]
     [StructLayout(LayoutKind.Sequential, Size = 16)]
-    public struct Vector128<T> where T : struct {}
+    public struct Vector128<T> where T : struct
+    {
+        // These fields exist purely so debug view works (https://github.com/dotnet/coreclr/issues/15694)
+        private byte _00;
+        private byte _01;
+        private byte _02;
+        private byte _03;
+        private byte _04;
+        private byte _05;
+        private byte _06;
+        private byte _07;
+        private byte _08;
+        private byte _09;
+        private byte _0A;
+        private byte _0B;
+        private byte _0C;
+        private byte _0D;
+        private byte _0E;
+        private byte _0F;
+
+        private unsafe string DisplayString
+        {
+            get
+            {
+                // The IsPrimitive check ends up working for `bool`, `char`, `IntPtr`, and `UIntPtr`
+                // which are not actually supported by any current architecture. This shouldn't be
+                // an issue however and greatly simplifies the check
+
+                if (typeof(T).IsPrimitive)
+                {
+                    var items = new T[16 / Marshal.SizeOf<T>()];
```

Comment:
Why not `Unsafe.SizeOf<T>`? `Marshal.SizeOf` has a different opinion about the size of `bool` so the line below will just do an unnecessary memory corruption, should the user use this with a pointless type.

Reply:
Will fix.

Reply:
Fixed.